geosnap/visualize/tsplot.py

```diff
@@ -0,0 +1,81 @@
+"""
+Visualization method for time series plots.
+"""
+
+import proplot as plot
+import contextily as ctx
+
+__all__ = ["tsplot"]
+
+"""
+Function for plotting timeseries data from community objects.
+
+Parameters
+----------
+community    : community object
+               community object
+column       : str
+               column to be graphed in a time series
+title        : str, optional
+               desired title of figure
+years        : list, optional
+               years to be graphed
+               default is every year in dataframe.
+scheme       : string,optional
+               matplotlib scheme to be used
+               default is 'quantiles'
+k            : int, optional
+               number of bins to graph.
+               Default is 5.
+save_fig     : boolean, optional
+               whether to save figure. Default is False.
+dpi          : int, optional
+               dpi of the saved image if save_fig=True
+               default is 500
+legend_kwds  : dictionary, optional
+               parameters for the legend
+               Default is 1 column on the bottom of the graph.
+ctxmap       : contextily map provider, optional
+               contextily basemap. Set to False for no basemap.
+               Default is OpenStreetMap.Mapnik       
+"""
+
+
+def tsplot(community, column, title='',
+           years=[], scheme='quantiles',
+           k=5,save_fig=False, dpi=500,
+           legend_kwds='default',
+           ctxmap=ctx.providers.OpenStreetMap.Mapnik,
+           **kwargs):
+    if legend_kwds == 'default':
+        legend_kwds = {'ncols': 1, "loc": "b"}
+    if ctxmap:  # need to convert crs to mercator before graphing
+        community.gdf = community.gdf.to_crs(epsg=3857)
+    if not years:
+        f, axs = plot.subplots(ncols=len(community.gdf.year.unique()))
+        for i, year in enumerate(sorted(community.gdf.year.unique())):  # sort to prevent graphing out of order
+            community.gdf[community.gdf.year == year].plot(column=column, ax=axs[i], scheme=scheme, k=k, **kwargs,
```

Comment:
this is looking really good! 

i only know this from prior experience, but one final thing we might want to consider is how to handle classifiers where `k` is endogenous, like max-p or maximum breaks in which case passing `k` will raise an error. In those cases we probably have to use an if/else since there are only a hadful of classifiers where this is an issue.

see [this section](https://github.com/spatialucr/geosnap/blob/de3ca3590da0eb71c24d348e94d36420ff878496/geosnap/visualize/commviz.py#L337) of the interactive community visualizer for an example. To be fair though, those classifiers are used fairly rarely in practice, so i'd be ok with merging this now and handling the edge cases later if you'd prefer to move onto something else

Reply:
I did some testing to see how the classifiers mentioned in `commviz` interacted with `k`. In my testing with this function, `maximum_breaks` was responsive to whatever `k` was set to, with `headtailbreaks` ignoring `k` completely. `maxp` took more than 5 minutes to run so I interrupted it. I didn't get any errors.

This testing suggests that mentioning these pitfalls in the docstring is probably an appropriate course of action for right now, like we discussed earlier.